account_invoice_triple_discount/migrations/16.0.2.0.0/post-migrate.py

```diff
@@ -15,7 +15,7 @@ def compute_discount(env):
             ("discount3", "!=", 0),
         ]
     )
-    env["account.move.line"]._compute_discount()
+    env["account.move.line"].filtered(lambda l:  l.date > l.company_id.fiscalyear_lock_date)._compute_discount()
```

Comment:
IMO here we either need to use a context key to bypass the restriction on lock date or bypass the ORM completely to do the update, otherwise all documents before lock date will end up showing a wrong discount.

Reply:
@grindtildeath  i am not following, do you think about some particular ctxt key which is already in the code?